src/odemis/driver/test/xt_client_test.py

```diff
@@ -1170,6 +1122,50 @@ def test_beam_powerVA(self):
         self.assertTrue(self.scanner.power.value)
         self.scanner.power.value = init_beam_power
 
+    def test_apply_autostigmator(self):
+        """
+        Test for the auto stigmation functionality.
+        """
+        detector = 'se-detector'  # autostigmator only works for the se-detector with channel=electron1
+        channel = DETECTOR2CHANNELNAME[detector]
+
+        # Run autostigmator flash script
+        autostigmator_future = self.scanner.applyAutoStigmator(detector)
+        # Don't check .is_autostigmating, function uses the same lock as .applyAutoStigmator, so it's only
+        # ever called after the autoStigmator function finished. TODO: is that intended?
+        self.assertIsInstance(autostigmator_future, ProgressiveFuture)
+        autostigmator_future.result()
+        autostigmator_state = self.microscope.is_autostigmating(channel)
+        self.assertEqual(autostigmator_state, False)
+
+        # TODO: Commented out, because cancelling is not an option for now.
+        # # Stop auto stigmation and check if it stopped running.
+        # autostigmator_future.cancel()
+        # time.sleep(5.0)  # Give microscope/simulator the time to update the state
+        # autostigmator_state = self.microscope.is_autostigmating(channel)
+        # self.assertEqual(autostigmator_state, False)
+
+        max_execution_time = 30  # Approximately 3 times the normal expected execution time (s)
+        for i in range(0, 5):
```

Comment:
Why repeat this 5 times? 

Also: you can just write `for i in range(5):`

Reply:
I don't know, I copied the testcase from you... I thought there was a reason that it might be a good idea to test multiple times, but maybe we can delete this entire part.

Reply:
Kornee originally wrote this code, @Kleijwegt can we delete this?

Reply:
I assume I put it there for a reason, I think the reason is that the autostigmation process wasn't reliable and sometimes didn't properly finish etc. I don't now what the current state is at the microscope. (of course on the simulator this won't be a problem)

Reply:
It looked to me that it was mostly there to get an idea how long the function takes on average. I deleted it in the new testcase.